fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintenanceTest.java

```diff
@@ -900,6 +904,137 @@ void sampledDelete(boolean isSynthetic, boolean isGrouped, long seed) throws IOE
                 assertThat(partitionCounts, Matchers.contains(5, 3, 4)));
     }
 
+    static Stream<Arguments> removeEmptyPartitions() {
```

Comment:
Would it be valuable to have a test that, in a loop:
1. deletes random records
2. runs merge
3. validates partition info, including that there are no empty merges
4. repeats until all records are deleted, at which point there should be one empty partition